e2e/wdio/core/tests/date-picker.e2e-spec.ts

```diff
@@ -0,0 +1,387 @@
+
+    it('verify null validity date-picker', () => {
+        click(activeButtonDatePicker, nullValidityIndex);
+        click(dayInCalendarButtonByValue('1'));
+        expect(getAttributeByName(activeInputDatePicker, 'ng-reflect-model', 6)).toEqual(date);
+    });
+
+    it('verify date picker in reactive form', () => {
+        click(activeButtonDatePicker, reactiveFormIndex);
+        click(dayInCalendarButtonByValue('1'));
+        expect(getAttributeByName(activeInputDatePicker, 'ng-reflect-model', 7)).toEqual(date);
+    });
+
+    it('verify date picker in reactive form with a range of dates', () => {
+        click(activeButtonDatePicker, reactiveFormRoDIndex);
+        click(dayInCalendarButtonByValue('1'));
+        click(dayInCalendarButtonByValue('15'));
+        expect(getAttributeByName(activeInputDatePicker, 'ng-reflect-model', 8)).toEqual(reactiveFormRoDDate);
+    });
+
+    it('verify date picker reactive form with disable function', () => {
+        click(activeButtonDatePicker, reactiveFormDisableIndex);
+        click(dayInCalendarButtonByValue(new Date().getDate().toString()));
+        expect(getAttributeByName(activeInputDatePicker, 'ng-reflect-model', 9)).toEqual(reactiveFormDisableDate);
+    });
+
+    it('verify date picker range in reactive form with disable functions.', () => {
+        click(activeButtonDatePicker, reactiveFormDisableRoDIndex);
+        click(dayInCalendarButtonByValue(new Date().getDate().toString()));
+        expect(getAttributeByName(activeInputDatePicker, 'ng-reflect-model', 10)).toEqual(reactiveFormDisableRoDDate);
+
+    });
+
+    it('verify date picker position', () => {
+        click(activeButtonDatePicker, positionIndex);
+        click(dayInCalendarButtonByValue('1'));
+        expect(getAttributeByName(activeInputDatePicker, 'ng-reflect-model', 11)).toEqual(date);
+        click(activeButtonDatePicker, positionIndex);
+    });
+
+    it('verify with the date picker, the user can see a day view, month view, year view, or year ranges.', () => {
+        const buttonsLength = getElementArrayLength(activeButtonDatePicker);
+        for (let i = 1; i < buttonsLength; i++) {
+            click(activeButtonDatePicker, i);
+            waitForElDisplayed(filterCalendarValue('day'));
+            click(buttonSelectMonth);
+            waitForElDisplayed(filterCalendarValue('month'));
+            click(buttonSelectYear);
+            waitForElDisplayed(filterCalendarValue('year'));
+            click(buttonSelectYearsRange);
+            waitForElDisplayed(filterCalendarValue('aggregated-year'));
+            click(activeButtonDatePicker, i);
+        }
+    });
+
+    it('verify selected date is showing in blue background', () => {
+        click(activeButtonDatePicker);
+        click(dayInCalendarButtonByValue('1'));
+        expect(getAttributeByName(activeInputDatePicker, 'ng-reflect-model')).toEqual(date);
+        click(activeButtonDatePicker);
+        expect(highlightedColor)
```

Comment:
testing against hardcoded color should not be done

Reply:
We'll discuss this point on our meeting on Thursday and and will fix it if necessary after it